.github/workflows/ci.yml

```diff
@@ -87,6 +87,56 @@ jobs:
         if: steps.cache.outputs.cache-hit != 'true' && matrix.cache == 'true'
         run: rm -rf ~/.m2/repository/io/trino/trino-*
 
+  reproducible-build:
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0 # checkout all commits, as the build result depends on `git describe` equivalent
+          ref: |
+            ${{ github.event_name == 'repository_dispatch' &&
+                github.event.client_payload.pull_request.head.sha == github.event.client_payload.slash_command.args.named.sha &&
+                format('refs/pull/{0}/head', github.event.client_payload.pull_request.number) || '' }}
+      - uses: ./.github/actions/setup
+        with:
+          cache: 'false'
```

Comment:
Why not `restore`?

Reply:
I want a clean build without any dependencies downloaded

Reply:
Why? in this job, we invoke `mvn clean install` twice, but we don't prune `~/.m2/repository` between the builds, so we could as well pull stuff from the cache.

Reply:
We are passing local repository - we are not using .m2/repository on that builds

Reply:
add a code comment also before `cache: 'false'` line